### PR TITLE
[CELEBORN-1699] Add GA to check the openapi codegen

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?
Check the consistency between openapi spec and generated openapi-client code in GA.
Fix the ThreadStack model.

### Why are the changes needed?
Address comments: https://github.com/apache/celeborn/pull/2892#issuecomment-2463170494

Followup for https://github.com/apache/celeborn/pull/2888
### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?

GA.

It works:
https://github.com/apache/celeborn/actions/runs/11733693060/job/32688436233?pr=2893

image

After ThreadStack fixed:
image

### CODE REVIEW

**Comment by turboFei** [comment]
https://github.com/apache/celeborn/actions/runs/11733693060/job/32688436233?pr=2893

image

Works as expected, cc @SteNicholas

**Comment by turboFei** [comment]
Maven command works as well:
```
build/mvn -pl openapi/openapi-client clean package -Pcheck
```

image

**Review by SteNicholas — APPROVED** [review]

File: openapi/openapi-client/src/main/openapi3/master_rest_v1.yaml [hunk 1]
```diff
@@ -583,6 +583,13 @@ components:
         inNative:
           type: boolean
           description: Whether the thread is executing native code.
+        isDaemon:
```
Comment (SteNicholas) [inline openapi/openapi-client/src/main/openapi3/master_rest_v1.yaml:586]: Remove `isDaemon` and `priority` and change `ThreadStack` with removing the fields.

Comment (turboFei) [reply openapi/openapi-client/src/main/openapi3/master_rest_v1.yaml]: ok, done

**Review by SteNicholas — APPROVED** [review]

**Comment by SteNicholas** [comment]
@turboFei, thanks for update. Merged to main(v0.6.0).

**Pull request closed without merging**